bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java

```diff
@@ -236,18 +236,22 @@ public String getKey() {
     }
 
     @Override
-    public void storeKey(@Nullable String key) {
-        // store it current configuration and avoiding complete re-initialization via handleConfigurationUpdate
-        getLGWebOSConfig().key = key;
-
-        // persist the configuration change
-        Configuration configuration = editConfiguration();
-        configuration.put(LGWebOSBindingConstants.CONFIG_KEY, key);
-        updateConfiguration(configuration);
+    public synchronized void storeKey(@Nullable String key) {
+        if (!getKey().equals(key)) {
+            logger.debug("store new key");
+            // store it current configuration and avoiding complete re-initialization via handleConfigurationUpdate
+            getLGWebOSConfig().key = key;
+
+            // persist the configuration change
+            Configuration configuration = editConfiguration();
+            configuration.put(LGWebOSBindingConstants.CONFIG_KEY, key);
+            updateConfiguration(configuration);
+        }
     }
 
     @Override
-    public void storeProperties(Map<String, String> properties) {
+    public synchronized void storeProperties(Map<String, String> properties) {
```

Comment:
why synchronized?

Reply:
This is not the case with the current code but with another of my changes, you could have different threads calling storeProperties at the same time.
By the way, I see that there is an object synchronizetion inside updateProperties and updateConfiguration so this is very probably useless to add one at the binding level.